### PR TITLE
Refactored code to reconnect after a disconnect

### PR DESCRIPTION
**Pull request by peterbarrow**
Detect when the node is uncontactable and pause trading until we are able to get all the connections back again.

### CODE REVIEW

File: bot/normal/normal.go [hunk 1]
```diff
@@ -225,6 +229,13 @@ func (b *Bot) submitLiquidityProvision(sub *api.PrepareLiquidityProvisionRequest
 	return nil
 }
 
+func (b *Bot) canPlaceTrades() bool {
+	if b.marketData.MarketTradingMode == proto.Market_TRADING_MODE_CONTINUOUS {
+		return true
+	}
+	return false
+}
```
Comment (ashleyvega) [inline bot/normal/normal.go:237]: --> `return  b.marketData.MarketTradingMode == proto.Market_TRADING_MODE_CONTINUOUS` perhaps

File: bot/normal/normal.go [hunk 2]
```diff
@@ -292,18 +303,20 @@ func (b *Bot) checkForShapeChange() {
 	if (b.openVolume > 0 && b.previousOpenVolume <= 0) ||
 		(b.openVolume < 0 && b.previousOpenVolume >= 0) {
 
+		b.log.Debugln("Flipping LP direction:", shape)
```
Comment (ashleyvega) [inline bot/normal/normal.go:306]: Suggestion:

```go
b.log.WithFields(log.Fields{"shape": shape}).Debug("Flipping LP direction")
```

File: bot/normal/normal.go [hunk 3]
```diff
@@ -158,6 +158,10 @@ func (b *Bot) Start() error {
 	b.stopPosMgmt = make(chan bool)
 	b.stopPriceSteer = make(chan bool)
 
+	err = b.initialiseData()
+	if err != nil {
+		return err
```
Comment (ashleyvega) [inline bot/normal/normal.go:163]: --> `return fmt.Errorf("failed to initialise data: %w", err)`

**Review by ashleyvega — APPROVED** [review]
LGTM

File: bot/normal/normal.go (hunk 2 above)
Comment (peterbarrow) [reply bot/normal/normal.go]: Fixed, thanks.

File: bot/normal/normal.go (hunk 3 above)
Comment (peterbarrow) [reply bot/normal/normal.go]: Fixed, thanks.

File: bot/normal/normal.go (hunk 1 above)
Comment (peterbarrow) [reply bot/normal/normal.go]: Fixed, thanks.

**Pull request merged**